overlord/snapstate/snapmgr.go

```diff
@@ -556,6 +556,29 @@ func (m *SnapManager) RefreshSchedule() (string, bool, error) {
 	return m.autoRefresh.RefreshSchedule()
 }
 
+// EnsureAutoRefreshesAreDelayed will delay refreshes for the specified amount
+// of time, as well as return any active auto-refresh changes that are currently
+// not ready so that the client can wait for those.
+func (m *SnapManager) EnsureAutoRefreshesAreDelayed(delay time.Duration) ([]*state.Change, error) {
+	// always delay for at least the specified time, this ensures that even if
+	// there are active refreshes right now, there won't be more auto-refreshes
+	// that happen after the current set finish
+	err := m.autoRefresh.ensureRefreshHoldAtLeast(delay)
+	if err != nil {
+		return nil, err
+	}
+
+	// look for auto refresh changes in progress
+	autoRefreshChgs := []*state.Change{}
```

Comment:
(nitpick) maybe use something like "active", "in-flight" or similar in the var name? e.g. `autoRefreshChgsInFlight` or so?